mitodl/ocw-hugo-themes
# course drawer state maintained

#### Pre-Flight checklist


#### What are the relevant tickets?
Closes https://github.com/mitodl/ocw-hugo-themes/issues/795

#### What's this PR do?
- Maintains desktop course drawer state across different course pages.
- **`Approach`**:
  - Whenever the user, either closes or opens the course drawer, its state (`opened` or `closed`) is stored in local storage.
  - Whenever any page is loaded, a function is triggered to check if there is any previously stored drawer state in local storage:
      - If yes, then it either closes or opens the drawer depending upon that state.
      - If not, then it sets the state to `opened` and opens the drawer by default 

#### How should this be manually tested?
- Checkout this branch.
- Build any course locally.
- Go to any page of that course (other than homepage)
- Verify that the course drawer is opened.
- Verify that the course drawer state (either opened or closed) is maintained across different course pages.
- One approach:
    - Close the drawer.
        - Either refresh the page or go to any other page and verify that the drawer is still closed. 
        - Close the tab, open a new tab and verify that the drawer is still closed.
    - Open the drawer and repeat the above steps again and now the drawer should remain open.
- Test this on multiple browsers (chrome, safari, firefox etc)
- Now block the cookies in your browser settings, test the flow again and verify that no error is logged in sentry, only an error message will be shown in console. (functionality will not work with blocked/disabled cookies)
- Test any other use-case that needs to be tested.  

#### Any background context you want to provide?
- Currently this branch is created from `abdurrehman/compact-design` branch instead of `main`, once compact design PR is merged then we can change the base of this PR
